src/main/java/com/philips/swcoe/cerberus/cerebellum/codemetrics/java/AbstractCodeMetricsWriterService.java

```diff
@@ -12,55 +15,67 @@
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-public class AbstractCodeMetricsWriterService  {
+public class AbstractCodeMetricsWriterService {
 
     protected List<String> classConfig;
     protected List<String> methodConfig;
     protected StringWriter reportWriter;
-    protected char delimiter;
+    protected String format;
     protected CSVPrinter csvPrinter;
+    protected Builder markdownPrinter;
 
-    public AbstractCodeMetricsWriterService(List<String> classConfig, List<String> methodConfig, char delimiter) throws IOException {
+    public AbstractCodeMetricsWriterService(List<String> classConfig, List<String> methodConfig, String format) throws IOException {
         reportWriter = new StringWriter();
-        this.delimiter = delimiter;
+        this.format = format;
         this.classConfig = classConfig;
         this.methodConfig = methodConfig;
     }
 
-
     protected Stream<Method> getStreamMetricsMethods(Method... methods) {
-        Stream<Method> methodStream = Arrays.stream(methods).filter(method -> method.getReturnType().equals(CodeMetricsResult.class));
+        Stream<Method> methodStream = Arrays.stream(methods).filter(method -> method.getReturnType().equals(CodeMetricsDiffResult.class));
         return methodStream;
     }
 
-    protected boolean doesItMatterToDisplay(CodeMetricsResult codeMetricsResult, List<String> methodMetricsToDisplay) {
-        Boolean doesItMatter = codeMetricsResult.getOldValue() != codeMetricsResult.getNewValue();
-        return shouldWeDisplay(methodMetricsToDisplay, codeMetricsResult) && doesItMatter;
+    protected boolean doesItMatterToDisplay(CodeMetricsDiffResult codeMetricsDiffResult, List<String> methodMetricsToDisplay) {
```

Comment:
I had same cognitive load to follow the logic. Especially ambiguous is that listOfMetricsToDisplay.isEmpty really means show all metrics. Alternatives include populating this list if user does not specify (i.e., show all) or have wrapper class for metrics to display instead of raw list, then ask it, i.e., "metrics.shouldDisplay(string name)" rather than a different class containing this display logic.

Reply:
I do not think performance is an issue for this application (yet) but you do not need to check both. If either fails, the metric will not be shown. You can make a composite boolean and use Java short-circuit processing order for && to deal with only running the second check if needed. For example: return metrics.shouldDisplay(metric) && mertic.hasChanged()). Three benefits: lower cognitive load and logic for what should display is in its natural home (the configured metrics list) and logic for metric has changed is in its natural home (the metric). Right now you need to pull the data out of the metric result object to do the conditional. "Feature envy" maybe. Depends if you need metric result class to be pure data for another reason.

Reply:
Thanks @dsvellal You are correct these names are really not so correct and creates a confusion . I have now changed it please check .

Reply:
@simaos thanks for your comments, you are correct I have made it as OR since what ever turns true all conditions get evaluated, and if either gets evaluated to true, I am rendering the metrics .  You are right I am using metric result as pure data object.